### ISSUE TITLE
[gmp] Build error on arm64-android

### ISSUE THREAD

**Issue by bmanga** [microsoft/vcpkg#36087]
Package: gmp:arm64-android@6.2.1#21

**Host Environment**

- Host: x64-linux
- Compiler: Clang 17.0.2
-    vcpkg-tool version: 2023-12-12-1c9ec1978a6b0c2b39c9e9554a96e3e275f

I think the relevant part is:
```
checking for aarch64-linux-android-gcc... [...]/Android/Sdk/ndk/android-ndk-r26b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
checking whether the C compiler works... no
```

 [issue_body.md](https://github.com/microsoft/vcpkg/files/13872515/issue_body.md)

**Comment by dg0yt** [comment]
Android NDK r26 can't be used for auto-tools based ports in vcpkg at the moment. Toolchain customization (to use prefixed clang) might work, or (a subset of) the changes proposed in #31228. Cf. #35851.

**Comment by LilyWangLL** [comment]
Closed as duplicate of #35960.
